### ISSUE TITLE
JaVers dependency on Google Guava

### ISSUE THREAD

**Issue by shirishpandharikar** [javers/javers#692]
As documented I just ran the below code to view the diff. It seems that Google Guava is required as a runtime dependency for the below code to work.

```java
Javers javers = JaversBuilder.javers().build();
Person tommyOld = new Person("tommy", "Tommy Smart");
Person tommyNew = new Person("tommy", "Tommy C. Smart");

Diff diff = javers.compare(tommyOld, tommyNew);
System.out.println(diff.prettyPrint());
```
The code above throws an exception as shown below:

```
Exception in thread "main" java.lang.NoClassDefFoundError: com/google/common/collect/ImmutableMap
	at org.javers.core.Changes.<init>(Changes.java:35)
	at org.javers.core.diff.Diff.groupByObject(Diff.java:88)
	at org.javers.core.diff.Diff.toString(Diff.java:129)
	at org.javers.core.diff.Diff.prettyPrint(Diff.java:120)
	...
Caused by: java.lang.ClassNotFoundException: com.google.common.collect.ImmutableMap
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 5 more
```

Not sure how the below line of code is used in [org.javers.core.Changes.java](https://github.com/javers/javers/blob/d5ac085c94b5459d464a14ab3e489f31e3c7c110/javers-core/src/main/java/org/javers/core/Changes.java#L35)
There are 3 more references in other classes HibernateConfig.java, JaversSpringJpaApplicationConfig.java and JaversSpringMongoApplicationConfig.java

The above code can be easily replaced with standard Java code to remove dependency on Goolge Guava.

Do let me know if these changes can be done. I will submit a PR.

**Comment by bartoszwalacik** [comment]
Thanks for reporting. You are right, there should be no references to Guava in javers-core classes (except of `org.javers.guava` package). Guava is the optional dependency:

```
optional "com.google.guava:guava:$guavaVersion"
```
and should not be required. I will be happy if you contribute a PR to fix it.

**Comment by bartoszwalacik** [comment]
released in 3.11.0

**Comment by PodusovMaxim** [comment]
Hi! 
I have same problem with guava stream:
`
java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.javers.common.reflection.ReflectionUtil.newInstance(ReflectionUtil.java:89)
.. 64 more
Caused by: java.lang.NoClassDefFoundError: com/google/common/collect/Streams
	at org.javers.common.collections.Primitives.getPrimitiveAndBoxTypes(Primitives.java:30)
	at org.javers.core.metamodel.type.TypeMapper.registerCoreTypes(TypeMapper.java:54)
	at org.javers.core.metamodel.type.TypeMapper.<init>(TypeMapper.java:40)
... 69 more
`

**Comment by shirishpandharikar** [comment]
@PodusovMaxim What version are you using? I looked inside the `org.javers.common.collections.Primitives` class. I do not see any reference to com/google/common/collect/Streams class. `getPrimitiveAndBoxTypes` methods just returns a statically defined array of classes.

**Comment by shirishpandharikar** [comment]
@PodusovMaxim Thanks for reporting this. This is something that got added after #692 was reported and fixed. Will submit a PR.

**Comment by bartoszwalacik** [comment]
finally fixed in javers-3.11.1

**Comment by tWechdorn** [comment]
RealNodePair.java also has a dependency on com.google.common.collect.Streams => which leads to a java.lang.NoClassDefFoundError: com/google/common/collect/Streams  in getPropertiesFromBothSides()

**Comment by shirishpandharikar** [comment]
@tWechdorn The changes to `RealNodePair` were done much later than this fix. I will fix the code and submit a PR.

**Comment by bartoszwalacik** [comment]
fixed in 5.10.3
